controllers/installation_controller.go

```diff
@@ -189,6 +197,207 @@ func (r *InstallationReconciler) ReportInstallationChanges(ctx context.Context,
 	}
 }
 
+// HashForAirgapConfig generates a hash for the airgap configuration. We can use this to detect config changes between
+// different reconcile cycles.
+func (r *InstallationReconciler) HashForAirgapConfig(in *v1beta1.Installation) (string, error) {
+	data, err := json.Marshal(in.Spec.Artifacts)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal artifacts location: %w", err)
+	}
+	hash := fmt.Sprintf("%x", sha256.Sum256(data))
+	return hash[:10], nil
+}
+
+// CreateArtifactJobForNode creates a job to copy assets from the internal registry. This jobs runs in the provided node.
+func (r *InstallationReconciler) CreateArtifactJobForNode(ctx context.Context, in *v1beta1.Installation, node corev1.Node) error {
+	log := ctrl.LoggerFrom(ctx)
+	log.Info("Creating artifact job for node", "node", node.Name, "installation", in.Name)
+	raw, err := static.Assets.ReadFile("assets/artifacts-job.yaml")
+	if err != nil {
+		return fmt.Errorf("failed to read job template: %w", err)
+	}
+
+	var job batchv1.Job
+	if err := yaml.Unmarshal(raw, &job); err != nil {
+		return fmt.Errorf("failed to unmarshal job template: %w", err)
+	}
+
+	hash, err := r.HashForAirgapConfig(in)
+	if err != nil {
+		return fmt.Errorf("failed to hash airgap config: %w", err)
+	}
+
+	labels := map[string]string{
+		"embedded-cluster/installation":          in.Name,
+		"embedded-cluster/artifacts-config-hash": hash,
+	}
+	job.Name = fmt.Sprintf("copy-artifacts-%s", node.Name)
+	job.Spec.Template.Labels, job.Labels = labels, labels
+	job.Spec.Template.Spec.NodeName = node.Name
+	job.Spec.Template.Spec.Containers[0].Env = append(
+		job.Spec.Template.Spec.Containers[0].Env,
+		corev1.EnvVar{Name: "INSTALLATION", Value: in.Name},
+	)
+
+	if err := r.Create(ctx, &job); err != nil {
+		return fmt.Errorf("failed to create job: %w", err)
+	}
+	log.Info("Artifact job for node created", "node", node.Name, "installation", in.Name)
+	return nil
+}
+
+// CopyVersionMetadataToCluster makes sure a config map with the embedded cluster version metadata exists in the
+// cluster. The data is read from the internal registry on the repository pointed by EmbeddedClusterMetadata.
+func (r *InstallationReconciler) CopyVersionMetadataToCluster(ctx context.Context, in *v1beta1.Installation) error {
```

Comment:
I'm confused by this. Why are we storing the metadata as a configmap instead of just reading it into memory when it's requested and caching it like we do the other metadata items today?

Reply:
+1

Reply:
That would be an option. I was trying to avoid using the registry every time the process starts. Do you want me to keep it on memory ? I can do that but I prefer this way. Let me know.

Reply:
I'd rather hit the registry every time that the process starts / a new EC version is requested, honestly

If the registry is down it's not like we should be changing anything anyways

Reply:
As per agreed we won't be doing this as to have the metadata as a config map is handy, specially at install time.